File: examples/simple/JsonConsumer.rb
#!/usr/bin/env ruby
$LOAD_PATH.push '../../lib/rb'
require 'consumer'

c = Jp::JsonConsumer.new 'json' do |message|
	print "I consumed:\n%s\n" % message.inspect
	true # must return something that evalutes to true in a boolean context, or purge() isn't called.
end
c.run

File: examples/simple/JsonConsumer.php
#!/usr/bin/php
<?php

set_time_limit(0);

$jpRoot = dirname(__FILE__) . '/../../';
set_include_path(get_include_path() . PATH_SEPARATOR . $jpRoot . 'lib/php' . PATH_SEPARATOR . $jpRoot . 'gen-php/jp');
require_once('jp/consumer/Json.php');
require_once('jp/worker/Interface.php');

class JsonQueueWorker implements jp_worker_Interface {
	/**
	 * @param array $item
	 * @return void
	 */
	public function processItem($item) {
		echo 'I consumed: ' . print_r($item, true);
		return true;
	}
}

$processor = new jp_consumer_Json('json', array('poll_interval' => 200), new JsonQueueWorker());
$processor->run();
